### PR TITLE
velero vSphere plugin docs for v1.3.0

### PR DESCRIPTION
**Pull request by deepakkinni**
**What this PR does / why we need it**:
Updates the current docs.

**Which issue(s) this PR fixes**:
Fixes: #418 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```
Signed-off-by: Deepak Kinni <dkinni@vmware.com>

### CODE REVIEW

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/649/

File: docs/guest.md [hunk 1]
```diff
@@ -22,24 +22,40 @@ The Data Manager Virtual Machine also needs to be installed to enable TKGS backu
 
 1. [Install Velero](https://velero.io/docs/v1.5/basic-install/)
 2. [Install Object Storage Plugin](#install-object-storage-plugin)
-3. [Install Velero Plugin for vSphere](#install-velero-plugin-for-vsphere)
+3. [Create Velero vSphere Plugin Config](#create-velero-vsphere-plugin-config)
+4. [Install Velero Plugin for vSphere](#install-velero-plugin-for-vsphere)
```
Comment (xing-yang) [inline docs/guest.md:26]: Can the naming to consistent?
Either velero-plugin-for-vsphere or velero-vsphere-plugin

File: docs/guest.md [hunk 2]
```diff
@@ -22,24 +22,40 @@ The Data Manager Virtual Machine also needs to be installed to enable TKGS backu
...
 
 ### Install Object Storage Plugin
 
 Volume backups are stored in an object store bucket. They are stored in the same bucket configured for the object storage plugin of Velero. Before installing the vSphere plugin, a Velero object storage plugin is required.
 
 Currently, only AWS plugin is supported and compatible with vSphere plugin. Please refer to [velero-plugin-for-aws](https://github.com/vmware-tanzu/velero-plugin-for-aws/blob/master/README.md) for more details about using **AWS S3** as the object store for backups. S3-compatible object stores, e.g, **MinIO**, are also supported via AWS plugin. Please refer to [install with MinIO](https://velero.io/docs/v1.5/contributions/minio/).
 
+### Create Velero vSphere Plugin Config
+
+The config map indicates to the plugin the `GUEST` cluster flavor
```
Comment (xing-yang) [inline docs/guest.md:36]: Can you rephrase the 2nd part to clarify?

File: docs/guest.md [hunk 3]
```diff
@@ -50,6 +66,49 @@ Please refer to [velero-plugin-for-vsphere tags](https://github.com/vmware-tanzu
...
+The following command scales down the `velero` deployment and deletes `backup-driver`.
+
+```bash
+kubectl -n <velero-namespace> delete deploy/backup-driver
+kubectl -n <velero-namespace> scale deploy/velero --replicas=0
+```
+
+### Create Velero vSphere Plugin ConfigMap
+
+The config map indicates to the plugin the `GUEST` cluster flavor
```
Comment (xing-yang) [inline docs/guest.md:87]: Same comment

File: docs/vanilla.md [hunk 4]
```diff
@@ -35,32 +35,138 @@ of VDDK Programming Guide. **Note: please apply privileges at the vCenter Server
...
 ### Install Object Storage Plugin
 
 Volume backups are stored in an object store bucket. They are stored in the same bucket configured for the object storage plugin of Velero. Before installing the vSphere plugin, a Velero object storage plugin is required.
 
 Currently, only AWS plugin is supported and compatible with vSphere plugin. Please refer to [velero-plugin-for-aws](https://github.com/vmware-tanzu/velero-plugin-for-aws/blob/master/README.md) for more details about using **AWS S3** as the object store for backups. S3-compatible object stores, e.g, **MinIO**, are also supported via AWS plugin. Please refer to [install with MinIO](https://velero.io/docs/v1.5/contributions/minio/).
 
+### Create VC Credential Secret
+
+You can retrieve these values from Secret `vsphere-config-secret` in `vmware-system-csi` namespace for
+2.3 vSphere CSI driver OR Secret `vsphere-config-secret` in `kube-system` namespace for 2.2 vSphere CSI
```
Comment (xing-yang) [inline docs/vanilla.md:51]: 2.3 and later?
2.2 and earlier?

File: docs/vanilla.md [hunk 5]
```diff
@@ -35,32 +35,138 @@ of VDDK Programming Guide. **Note: please apply privileges at the vCenter Server
...
+driver
```
Comment (xing-yang) [inline docs/vanilla.md:52]: What if vSphere CSI Driver is deployed as a process?

File: docs/vanilla.md [hunk 6]
```diff
@@ -35,32 +35,138 @@ of VDDK Programming Guide. **Note: please apply privileges at the vCenter Server
...
+
+```bash
+% cat csi-vsphere.conf
+[Global]
+cluster-id = "cluster1"
+
+[VirtualCenter "10.182.1.133"]
+user = "Administrator@vsphere.local"
```
Comment (xing-yang) [inline docs/vanilla.md:60]: Can you use a fake user name?

File: docs/vanilla.md [hunk 7]
```diff
@@ -35,32 +35,138 @@ of VDDK Programming Guide. **Note: please apply privileges at the vCenter Server
...
+password = "password"
+port = "443"
+
+% kubectl -n <velero-namespace> create secret generic velero-vsphere-config-secret --from-file=csi-vsphere.conf
+```
+
+### Create Velero vSphere Plugin Config
+
+The config map indicates to the plugin the cluster flavor and VC Credential Secret created in the previous step
```
Comment (xing-yang) [inline docs/vanilla.md:69]: Rephrase a little to make is clear

File: docs/vanilla.md [hunk 8]
```diff
@@ -35,32 +35,138 @@ of VDDK Programming Guide. **Note: please apply privileges at the vCenter Server
...
-For Version 1.1.1 the command is
+For Version 1.3.0 the command is
 
 ```bash
-velero plugin add vsphereveleroplugin/velero-plugin-for-vsphere:v1.1.1
+velero plugin add vsphereveleroplugin/velero-plugin-for-vsphere:v1.3.0
 ```
 
-Please refer to [velero-plugin-for-vsphere tags](https://github.com/vmware-tanzu/velero-plugin-for-vsphere/tags) for correct tags for versions 1.1.1 and higher.
+Please refer to [velero-plugin-for-vsphere tags](https://github.com/vmware-tanzu/velero-plugin-for-vsphere/tags) for correct tags for versions 1.3.0 and higher.
```
Comment (xing-yang) [inline docs/vanilla.md:96]: Make this change in guest.md.

File: docs/vanilla.md [hunk 9]
```diff
@@ -35,32 +35,138 @@ of VDDK Programming Guide. **Note: please apply privileges at the vCenter Server
...
+kubectl -n <velero-namespace> delete deploy/backup-driver
+kubectl -n <velero-namespace> delete daemonset datamgr-for-vsphere-plugin
+kubectl -n <velero-namespace> scale deploy/velero --replicas=0
+```
+
+### Create vSphere Credentials Secret
+
+You can retrieve these values from Secret `vsphere-config-secret` in `vmware-system-csi` namespace for
+2.3 vSphere CSI driver OR Secret `vsphere-config-secret` in `kube-system` namespace for 2.2 vSphere CSI
+driver
```
Comment (xing-yang) [inline docs/vanilla.md:126]: See similar comments earlier.

File: docs/vanilla.md [hunk 10]
```diff
@@ -35,32 +35,138 @@ of VDDK Programming Guide. **Note: please apply privileges at the vCenter Server
...
+```bash
+kubectl -n velero set image deployment/velero velero-plugin-for-vsphere=vsphereveleroplugin/velero-plugin-for-vsphere:v1.3.0
+```
+
+### Scale up Velero deployment
+
+```bash
+kubectl -n <velero-namespace> scale deploy/velero --replicas=1
+```
+
```
Comment (xing-yang) [inline docs/vanilla.md:169]: Do we need to say something about Supervisor in the supervisor doc section?

**Review by lintongj — COMMENTED** [review]
A general comment here. I would suggest to add a separate page for the detail of creating the ConfigMap with credential. In the cluster flavor specific landing page, let's add one entry into the list of prerequisites with EMPHASIS.

File: docs/vanilla.md (hunk 5 above)
Comment (deepakkinni) [reply docs/vanilla.md]: Added a line stating that they need to contact vsphere admin to retrieve the info in other cases.

File: docs/vanilla.md (hunk 10 above)
Comment (deepakkinni) [reply docs/vanilla.md:173]: no changes are necessary for supervisor install.

**Comment by deepakkinni** [comment]
> A general comment here. I would suggest to add a separate page for the detail of creating the ConfigMap with credential. In the cluster flavor specific landing page, let's add one entry into the list of prerequisites with EMPHASIS.

Actually, the ConfigMaps are different for guest and vanilla (no vc credentials secret necessary for guest), so having a common config map page wouldn't be possible.

**Comment by lintongj** [comment]
> > A general comment here. I would suggest to add a separate page for the detail of creating the ConfigMap with credential. In the cluster flavor specific landing page, let's add one entry into the list of prerequisites with EMPHASIS.
> 
> Actually, the ConfigMaps are different for guest and vanilla (no vc credentials secret necessary for guest), so having a common config map page wouldn't be possible.

Just maintain different sections in the separate page for vanilla and guest cluster flavor respectively. Otherwise, if put too many detail in the per-cluster landing page, users are easy to get lost.

**Comment by svcbot-qecnsdp** [comment]
Congratulations !! 

All the pre-check tests passed
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/657/

**Comment by lintongj** [comment]
> > > A general comment here. I would suggest to add a separate page for the detail of creating the ConfigMap with credential. In the cluster flavor specific landing page, let's add one entry into the list of prerequisites with EMPHASIS.
> > 
> > 
> > Actually, the ConfigMaps are different for guest and vanilla (no vc credentials secret necessary for guest), so having a common config map page wouldn't be possible.
> 
> Just maintain different sections in the separate page for vanilla and guest cluster flavor respectively. Otherwise, if put too many detail in the per-cluster landing page, users are easy to get lost.

@deepakkinni I took one another look at this PR. Ignore my previous comments. Generally, it looks good to organize this newly added section this way. 

My only comment here is, would it be better to also add a corresponding section to supervisor.md and call out that it is not required to create such a ConfigMap for plugin in supervisor cluster?

File: docs/guest.md [hunk 11]
```diff
@@ -50,6 +66,51 @@ Please refer to [velero-plugin-for-vsphere tags](https://github.com/vmware-tanzu
 
 If it is an air-gapped environment, please refer to [Install Notes with Customized Images](install-notes.md) for extra required steps.
 
+## Upgrade
```
Comment (lintongj) [inline docs/guest.md:69]: Would you please add an entry for `Upgrade` to  `Table of Contents`?

File: docs/vanilla.md [hunk 12]
```diff
@@ -35,32 +35,142 @@ of VDDK Programming Guide. **Note: please apply privileges at the vCenter Server
...
 ```
 
-Please refer to [velero-plugin-for-vsphere tags](https://github.com/vmware-tanzu/velero-plugin-for-vsphere/tags) for correct tags for versions 1.1.1 and higher.
+Please refer to [velero-plugin-for-vsphere tags](https://github.com/vmware-tanzu/velero-plugin-for-vsphere/tags) for correct tags for versions 1.3.0 and higher.
 
 #### Install in Air-gapped environment
 
 If it is an air-gapped environment, please refer to [Install Notes with Customized Images](install-notes.md) for extra required steps.
 
+## Upgrade
```
Comment (lintongj) [inline docs/vanilla.md:105]: Would you please add an entry for `Upgrade` to  `Table of Contents`?

File: docs/guest.md (hunk 11 above)
Comment (deepakkinni) [reply docs/guest.md:83]: Table of contents was missing here, added it.

**Comment by xing-yang** [comment]
LGTM

**Review by lintongj — APPROVED** [review]

**Comment by svcbot-qecnsdp** [comment]
Congratulations !! 

All the pre-check tests passed
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/658/
